src/main_gui.cpp

```diff
@@ -92,44 +92,48 @@ void CcPlaySound_EXPLOSION(Commands, const CommandCost &result, TileIndex tile)
  */
 bool DoZoomInOutWindow(ZoomStateChange how, Window *w)
 {
-	Viewport *vp;
-
 	assert(w != nullptr);
-	vp = w->viewport;
 
 	switch (how) {
 		case ZOOM_NONE:
 			/* On initialisation of the viewport we don't do anything. */
 			break;
 
-		case ZOOM_IN:
-			if (vp->zoom <= _settings_client.gui.zoom_min) return false;
-			vp->zoom = (ZoomLevel)((int)vp->zoom - 1);
-			vp->virtual_width >>= 1;
-			vp->virtual_height >>= 1;
+		case ZOOM_IN: {
+			Viewport &vp = *w->viewport;
+			if (vp.zoom <= _settings_client.gui.zoom_min) return false;
+			vp.zoom = (ZoomLevel)((int)vp.zoom - 1);
+			vp.virtual_width >>= 1;
+			vp.virtual_height >>= 1;
 
-			w->viewport->scrollpos_x += vp->virtual_width >> 1;
-			w->viewport->scrollpos_y += vp->virtual_height >> 1;
+			w->viewport->scrollpos_x += vp.virtual_width >> 1;
+			w->viewport->scrollpos_y += vp.virtual_height >> 1;
 			w->viewport->dest_scrollpos_x = w->viewport->scrollpos_x;
 			w->viewport->dest_scrollpos_y = w->viewport->scrollpos_y;
```

Comment:
Given the `vp = *w->viewport` above, why do we use `w->viewport` over `vp` here?

Reply:
Seems to be the difference between `Viewport` and `ViewportData`. I'm not sure why the second exists, as all instances are `ViewportData`, even if the viewport doesn't follow a vehicle. So maybe I should tidy that up.